pytest_socket.py

```diff
@@ -170,16 +170,52 @@ def host_from_connect_args(args):
         return host_from_address(address)
 
 
+def is_ipaddress(address: str):
+    """
+    Determine if the address is a valid IPv4 address.
+    """
+    try:
+        socket.inet_aton(address)
+        return True
+    except OSError:
+        return False
+
+
+def resolve_hostname(hostname):
+    try:
+        return socket.gethostbyname(hostname)
+    except socket.gaierror:
+        return None
+
+
+def treat_allowed(allowed):
+    allowed_hosts = []
+    for allow in allowed:
+        allow = allow.strip()
+        if is_ipaddress(allow):
+            allowed_hosts.append(allow)
+
+        resolved = resolve_hostname(allow)
+        if resolved:
+            allowed_hosts.append(resolved)
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**question:** ‏If I'm reading this correctly, then if the `allow` value is an IP address, then won't the IP appear twice in `allowed_hosts`, since the call to `gethostbyname()` will return the same IP value? Should we try to resolve an IP address, or skip once it's allowed, and reduce the duplication in the resulting output?

Reply:
Yes, I modified it to add an `else`, since it is either already an IP address or an attempt needs to be made to resolve it